src/include/socket.h

```diff
@@ -312,30 +312,110 @@ static int findInterfaces(char* ifNames, union socketAddress *ifAddrs, int ifNam
   return nIfs;
 }
 
+static ncclResult_t bindToPort(int sockfd, union socketAddress *localAddr, bool reuseport, bool *success) {
+  if (socketToPort(&localAddr->sa)) {
+    // Port is forced by env. Make sure we get the port.
+    int opt = 1;
+
+    int optname = SO_REUSEADDR;
+
+#if defined(SO_REUSEPORT)
+    if(reuseport){
+      optname |= SO_REUSEPORT;
+    }
+#endif
+
+    SYSCHECK(setsockopt(sockfd, SOL_SOCKET, optname, &opt, sizeof(opt)), "setsockopt");
+  }
+
+  int salen = (localAddr->sa.sa_family == AF_INET) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6);
+  int ret = bind(sockfd, &localAddr->sa, salen);
+  /* If caller provides non-NULL success, tolerate EADDRINUSE errors */
+  if (ret == -1 && errno == EADDRINUSE && success != NULL) {
```

Comment:
this divergent behavior based on `success` seems opaque; I can't think of a great solution but perhaps renaming success to `tolerateInUse` might help?

Reply:
I decided it was better to for the bindToPort to wrap *just* the call to bind, and pass the error checking responsibility entirely to the caller.